File: dart/com/spoonacular/client/model/inline_response20021.dart
part of openapi.api;

class InlineResponse20021 {
  
  InlineResponse20021Calories calories = null;
  
  InlineResponse20021Calories carbs = null;
  
  InlineResponse20021Calories fat = null;
  
  InlineResponse20021Calories protein = null;
  
  int recipesUsed = null;
  InlineResponse20021();

  @override
  String toString() {
    return 'InlineResponse20021[calories=$calories, carbs=$carbs, fat=$fat, protein=$protein, recipesUsed=$recipesUsed, ]';
  }

  InlineResponse20021.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['calories'] == null) {
      calories = null;
    } else {
      calories = new InlineResponse20021Calories.fromJson(json['calories']);
    }
    if (json['carbs'] == null) {
      carbs = null;
    } else {
      carbs = new InlineResponse20021Calories.fromJson(json['carbs']);
    }
    if (json['fat'] == null) {
      fat = null;
    } else {
      fat = new InlineResponse20021Calories.fromJson(json['fat']);
    }
    if (json['protein'] == null) {
      protein = null;
    } else {
      protein = new InlineResponse20021Calories.fromJson(json['protein']);
    }
    if (json['recipesUsed'] == null) {
      recipesUsed = null;
    } else {
          recipesUsed = json['recipesUsed'];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (calories != null)
      json['calories'] = calories;
    if (carbs != null)
      json['carbs'] = carbs;
    if (fat != null)
      json['fat'] = fat;
    if (protein != null)
      json['protein'] = protein;
    if (recipesUsed != null)
      json['recipesUsed'] = recipesUsed;
    return json;
  }

  static List<InlineResponse20021> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20021>() : json.map((value) => new InlineResponse20021.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20021> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20021>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20021.fromJson(value));
    }
    return map;
  }
}

File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20021.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20021Calories

/**
 * 
 * @param calories 
 * @param carbs 
 * @param fat 
 * @param protein 
 * @param recipesUsed 
 */
case class InlineResponse20021(calories: InlineResponse20021Calories,
                carbs: InlineResponse20021Calories,
                fat: InlineResponse20021Calories,
                protein: InlineResponse20021Calories,
                recipesUsed: Int
                )

object InlineResponse20021 {
    /**
     * Creates the codec for converting InlineResponse20021 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20021] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20021] = deriveEncoder
}
